File: commands/decline.js
const { SlashCommandBuilder } = require("discord.js");
const { findChallenge, completeChallenge } = require("../game-manager.js");
const { WordleGame } = require("../src/wordle.js");

module.exports = {
    data: new SlashCommandBuilder()
        .setName("decline")
        .setDescription("Decline a duel invitation against you"),
    async execute(interaction){
        // Create a new game for this channel if it doesnt already exist
        let challenge = findChallenge(interaction.user.id);
        if(challenge){
            completeChallenge(interaction.user.id, challenge.player2);
            interaction.reply({content: "Challenge declined!", ephemeral: true});
        } else
            interaction.reply({content: "There is no challenge against you!", ephemeral: true});
    },
};

File: commands/decline.go
package commands

import (
	"cordle/game"
	"fmt"

	"github.com/bwmarrin/discordgo"
)

// duelDecline declines a challenge against a given user
func duelDecline(s *discordgo.Session, i *discordgo.InteractionCreate){
	c := game.FindChallenge(i.Interaction.User)
	if c != nil{
		// Decline the challenge
		game.CloseChallenge(c)
		// Notify players that the challenge has been declined
		m := fmt.Sprintf(
			"%s, %s has declined your duel!",
			c.Target.Mention(),
			i.Interaction.User.Mention(),
		)
		respond(s, i, m, false)
	} else{
		respond(s, i, "You currently have no active challenges against you.", true)
	}
}